ocaml/xapi/repository_helpers.ml

```diff
@@ -875,38 +875,44 @@ let remove_repo_conf_file repo_name =
   in
   Unixext.unlink_safe path
 
-let write_yum_config ?(source_url = None) binary_url repo_name =
+let write_yum_config ?(source_url = None) ?(repo_gpgcheck = true) binary_url
+    repo_name gpgkey_name =
   let file_path =
     Filename.concat !Xapi_globs.yum_repos_config_dir (repo_name ^ ".repo")
   in
-  let gpgkey_path =
-    Filename.concat !Xapi_globs.rpm_gpgkey_dir
-      !Xapi_globs.repository_gpgkey_name
+  let opt_repo_gpgcheck, opt_gpgcheck, opt_gpgkey =
```

Comment:
This block seems quite convoluted, it would seem like matching values to construct the tuple would be more organized:

```ocaml
  let opt_gpgkey () =
    if not (Sys.file_exists gpgkey_path) then
        raise
          Api_errors.(
            Server_error (internal_error, ["gpg key file does not exist"])
          ) ;
    else if not ((Unix.lstat gpgkey_path).Unix.st_kind = Unix.S_REG) then
        raise
          Api_errors.(
            Server_error (internal_error, ["gpg key file is not a regular file"])
          ) ;
  else
      Printf.sprintf "gpgkey=file://%s" gpgkey_path
  in
  match !Xapi_globs.repository_gpgcheck, repo_gpgcheck with
  | false, _ ->
      ("repo_gpgcheck=0", "gpgcheck=0", "")
  | true, false ->
      ("repo_gpgcheck=0", "gpgcheck=1", opt_gpgkey ())
  | true, true ->
      ("repo_gpgcheck=1", "gpgcheck=1", opt_gpgkey ())
```